feat: add confirmation modal for delete action (#575)

JACTyt opened the pull request:
Hi, I created a PR that solves issue #575 .
Added delete_confirmation variable (true/false). 
If true, then there will be confirmation popup in the center of the screen with buttons Yes or No. 
If false, then no confirmation popup will appear. Simple deletion without confirmation (usage depends on user preferences).
Have some doubts about func  showModal() placement.

Closes #575

JACTyt committed:
Add confirmation window before message deletion

JACTyt committed:
Code polishing

ayn2op commented on cmd/application.go:
```diff
@@ -15,6 +16,7 @@ const (
 	flexPageName            = "flex"
 	mentionsListPageName    = "mentionsList"
 	attachmentsListPageName = "attachmentsList"
+	modalPageName           = "Modal"
```
Rename the const name to something more descriptive: `confirmModalPageName`:

```go
const confirmModalPageName = "confirmModal"
```

ayn2op commented on internal/config/config.toml:
```diff
@@ -70,6 +70,7 @@ reply = "Rune[r]"
 reply_mention = "Rune[R]"
 cancel = "Esc"
 delete = "Rune[d]"
+delete_confirmation = true
```
Perhaps, `Rune[d]` can be used for deleting messages with confirmation and `Rune[D]` for deleting them without. This is similar to what `reply` and `reply_mention`. What do you think?

ayn2op commented on internal/config/keys.go:
```diff
@@ -40,9 +40,10 @@ type (
-		Open   string `toml:"open"`
+		Cancel             string `toml:"cancel"`
+		Delete             string `toml:"delete"`
+		DeleteConfirmation bool   `toml:"delete_confirmation"`
```
https://github.com/ayn2op/discordo/pull/605/files#r2383657890

JACTyt committed:
Requested changes done

Changed delete to Rune[D] and delete_confirm to Rune[d](Much better). Renamed Modal into confirmModal.

ayn2op commented on cmd/application.go:
```diff
@@ -154,14 +154,14 @@ func (a *application) onFlexInputCapture(event *tcell.EventKey) *tcell.EventKey
 }
 
-func (a *application) showModal(prompt string, buttons []string, onDone func(label string)) {
+func (a *application) showConfirmModal(prompt string, buttons []string, onDone func(label string)) {
```
The `buttons` parameter can be removed and hardcoded to `[]string{"Cancel", "Confirm"}` instead.

JACTyt replied:
In my opinion, hardcoding this part is not a good idea. We can reuse confirmModal for example to provide some debug Information or when opening web pages (confirm if you trust the link and want to open it, etc.). I will do as you wish. Please verify, do i need to hardcode it?

ayn2op reviewed (approved):
LGTM! Thank you <3

ayn2op committed:
Merge branch 'main' into feat/delete_confirmation
